line/result.go

```diff
@@ -0,0 +1,103 @@
+package line
+
+import (
+	"bufio"
+	"io"
+	"strings"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/values"
+)
+
+// ResultDecoder decodes raw input strings from a reader into a flux.Result.
+// It uses a separator to split the input into tokens and generate table rows.
+// Tokens are kept as they are and put into a table with schema `_time`, `_value`.
+// The `_value` column contains tokens.
+// The `_time` column contains the timestamps for when each `_value` has been read.
+// Strings in `_value` are obtained from the io.Reader passed to the Decode function.
+// ResultDecoder outputs one table once the reader reaches EOF.
+type ResultDecoder struct {
+	reader *bufio.Reader
+	stats  flux.Statistics
+	config *ResultDecoderConfig
+}
+
+// NewResultDecoder creates a new result decoder from config.
+func NewResultDecoder(config *ResultDecoderConfig) *ResultDecoder {
+	return &ResultDecoder{config: config}
+}
+
+// TimeProvider gives the current time.
+type TimeProvider interface {
+	GetCurrentTime() values.Time
```

Comment:
nit, this should not need the prefix `Get`, so `CurrentTime` would be a better name for this method.